#! /bin/sh /usr/share/dpatch/dpatch-run
## curl-followlocation.dpatch by Florian Forster <octo@verplant.org>
##
## DP: Plugins using libcurl: Enable the ‘CURLOPT_FOLLOWLOCATION’ option.

@DPATCH@

diff a/src/apache.c b/src/apache.c
--- a/src/apache.c
+++ b/src/apache.c
@@ -458,6 +458,7 @@ static int init_host (apache_t *st) /* {{{ */
 	}
 
 	curl_easy_setopt (st->curl, CURLOPT_URL, st->url);
+	curl_easy_setopt (st->curl, CURLOPT_FOLLOWLOCATION, 1);
 
 	if (st->verify_peer != 0)
 	{
diff a/src/ascent.c b/src/ascent.c
--- a/src/ascent.c
+++ b/src/ascent.c
@@ -560,6 +560,7 @@ static int ascent_init (void) /* {{{ */
   }
 
   curl_easy_setopt (curl, CURLOPT_URL, url);
+  curl_easy_setopt (curl, CURLOPT_FOLLOWLOCATION, 1);
 
   if ((verify_peer == NULL) || (strcmp (verify_peer, "true") == 0))
     curl_easy_setopt (curl, CURLOPT_SSL_VERIFYPEER, 1);
diff a/src/bind.c b/src/bind.c
--- a/src/bind.c
+++ b/src/bind.c
@@ -1395,6 +1395,7 @@ static int bind_init (void) /* {{{ */
   curl_easy_setopt (curl, CURLOPT_USERAGENT, PACKAGE_NAME"/"PACKAGE_VERSION);
   curl_easy_setopt (curl, CURLOPT_ERRORBUFFER, bind_curl_error);
   curl_easy_setopt (curl, CURLOPT_URL, (url != NULL) ? url : BIND_DEFAULT_URL);
+  curl_easy_setopt (curl, CURLOPT_FOLLOWLOCATION, 1);
 
   return (0);
 } /* }}} int bind_init */
diff a/src/curl.c b/src/curl.c
--- a/src/curl.c
+++ b/src/curl.c
@@ -346,6 +346,7 @@ static int cc_page_init_curl (web_page_t *wp) /* {{{ */
       PACKAGE_NAME"/"PACKAGE_VERSION);
   curl_easy_setopt (wp->curl, CURLOPT_ERRORBUFFER, wp->curl_errbuf);
   curl_easy_setopt (wp->curl, CURLOPT_URL, wp->url);
+  curl_easy_setopt (wp->curl, CURLOPT_FOLLOWLOCATION, 1);
 
   if (wp->user != NULL)
   {
diff a/src/nginx.c b/src/nginx.c
--- a/src/nginx.c
+++ b/src/nginx.c
@@ -141,6 +141,8 @@ static int init (void)
     curl_easy_setopt (curl, CURLOPT_URL, url);
   }
 
+  curl_easy_setopt (curl, CURLOPT_FOLLOWLOCATION, 1);
+
   if ((verify_peer == NULL) || (strcmp (verify_peer, "true") == 0))
   {
     curl_easy_setopt (curl, CURLOPT_SSL_VERIFYPEER, 1);
